src/main/java/com/sap/oss/phosphor/fosstars/model/subject/oss/MavenArtifact.java

```diff
@@ -0,0 +1,99 @@
+package com.sap.oss.phosphor.fosstars.model.subject.oss;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import java.util.Optional;
+
+
+/**
+ * Maven artifact.
+ */
+public class MavenArtifact implements Artifact {
+
+  /**
+   * Maven artifact's group id.
+   */
+  private final String group;
+
+  /**
+   * Maven artifact's artifact id.
+   */
+  private final String artifact;
+
+  /**
+   * Maven artifact's GitHub project.
+   */
+  private final GitHubProject gitHubProject;
+
+  /**
+   * Initializes a maven artifact.
+   *
+   * @param group A group id of the artifact.
+   * @param artifact A artifact id.
+   * @param gitHubProject A {@link GitHubProject}.
+   */
+  @JsonCreator
+  public MavenArtifact(@JsonProperty("group") String group,
+      @JsonProperty("artifact") String artifact,
+      @JsonProperty("gitHubProject") GitHubProject gitHubProject) {
+    this.group = Objects.requireNonNull(group,
+        "Oh no! You gave me a null instead of a maven group from GA coordinate!");
+    this.artifact = Objects.requireNonNull(artifact,
+        "Oh no! You gave me a null instead of a maven artifact from GA coordinate!");
+    this.gitHubProject = gitHubProject;
+  }
+
+  /**
+   * Returns group id of the artifact.
+   *
+   * @return Group id.
+   */
+  @JsonGetter("group")
+  public String group() {
+    return group;
+  }
+
+  /**
+   * Returns artifact id.
+   *
+   * @return Artifact id.
+   */
+  @JsonGetter("artifact")
+  public String artifact() {
+    return artifact;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o instanceof MavenArtifact == false) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+    MavenArtifact that = (MavenArtifact) o;
+    return Objects.equals(group, that.group) && Objects.equals(artifact, that.artifact)
+        && Objects.equals(gitHubProject, that.gitHubProject);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), group, artifact, gitHubProject);
+  }
+
+  @Override
+  public String toString() {
+    return String.format("%s:%s %s", group, artifact, gitHubProject);
+  }
+
+  @Override
+  @JsonGetter("gitHubProject")
+  public Optional<GitHubProject> gitHubProject() {
```

Comment:
This getter returns an `Optional` but the constructor takes a `GitHubProject`. Did you test that serialization/deserialization works well? Could you please add a test for that? In general, I would be good to add unit tests for all new classes.